Scraper.py

```diff
@@ -6,164 +6,149 @@
-					# Scrape emails	and add them to self.emailList
-					self.emailList = self.Scrape_Emails(content, self.emailList)
-				
-				# Write to disk
-				if self.cycles % 100 == 0:
-					if self.emailList:
-						self.WriteToDisk()
-		
-				# Add url to donelist so we do not re-scrape
-				self.doneList.append(url)
-	
-				self.cycles += 1
-
-			except (KeyboardInterrupt, SystemExit):
-				print "\n\nKeyboardInterrupt caught, exiting..."
-				self.WriteToDisk()
-				shutil.rmtree(".cache")
-				self.shouldDie = True
-				sys.exit(0)
-	
-		try:
-			self.WriteToDisk()
-		except:
-			print "Unable to initially scrape the target; double-check domain name."
+    main_list = set() #list of fetched urls
+    email_list = set()
+    done_list = set()
+    bots = None
+    timeout = None
+    domain = None
+    lag = None
+    depth = None
+    should_die = False
+    depth_counter = 0
+    cycles = 0
+    h = httplib2.Http()
+
+    def __init__(self, domain=None,  bots=None, timeout=3, depth=3, lag=0.1):
+        self.domain = domain
+        self.main_list = [domain]
+        self.bots = bots
+        self.timeout = timeout
+        self.depth = depth
+        self.lag = lag
+        self.h = httplib2.Http(".cache", self.timeout)
+        self.h.follow_all_redirects = True
+        self.h.follow_redirects = True
+
+    def write_to_disk(self):
+        # Only write to a file if email_list list is not empty
+        if self.email_list:
+            print "Writing emails..."
+            file = open("emails.txt", "w")
+
+            for email in self.email_list:
+                file.write(email + "\n")
+            file.close()
+
+
+
+    def scrape_emails(self, webpage):
+        # Find all emails in a webpage using a regex
+        address =  sys.argv[1]
+        email_list = re.findall(r"/^[-_a-z0-9\'+*$^&%=~!?{}]+(?:\.[-_a-z0-9\'+*$^&%=~!?{}]+)*" + re.escape(address) + r"$/iD", webpage)
+
+        # Convert all of the found emails to lowercase
+        email_list= [email.lower() for email in email_list]
+
+        return set(email_list)
+
+
+    def scrape_links(self, soup, domain):
+        if domain.endswith("/"):
+            domain = domain[0:-1]
+
+        # For each link in the html code that has <a href= that is not blank (None) add it to url_list
+        url_list = [link.get("href") for link in soup.findAll("a") if link.get("href") is not None]
+
+        # First type of non-absolute links
+        nonAbsolute1 = [link[1:] for link in url_list if link.startswith("/")]
+        nonAbsolute1 = [domain + "/" + link for link in nonAbsolute1]
+
+        # Second type of non-absolute links
+        nonAbsolute2 = [domain + "/" + link for link in url_list if not link.startswith("http") and not link.startswith("/") and not link.startswith(".")]
+
+        url_list.extend(nonAbsolute1)
+        url_list.extend(nonAbsolute2)
+
+        # Only scrape first party domain
+        url_list = [url for url in url_list if url.startswith(domain)]

```

Comment:
if i'm reading it correctly in this instance, if url scraped was `http://jana.com`, and the original targeted domain was `http://jana.co` it would incorrectly match? (similar to `http://foo.jana.com` and `http://foo.jana.company.com`)